TypeCobol.LanguageServer/Workspace.cs

```diff
@@ -229,20 +219,18 @@ private FileCompiler OpenTextDocument(DocumentContext docContext, string sourceT
                 arrangedCustomSymbol.Programs.Remove(matchingPgm);
             }
 
-            fileCompiler = new FileCompiler(initialTextDocumentLines, CompilationProject.SourceFileProvider,
-                CompilationProject, CompilationProject.CompilationOptions, arrangedCustomSymbol ?? _customSymbols,
-                CompilationProject);
+            fileCompiler = new FileCompiler(initialTextDocumentLines, compilationProjet.SourceFileProvider,
+                compilationProjet, compilationProjet.CompilationOptions, arrangedCustomSymbol ?? _customSymbols,
+                compilationProjet);
 #else
-            fileCompiler = new FileCompiler(initialTextDocumentLines, CompilationProject.SourceFileProvider, CompilationProject, CompilationProject.CompilationOptions, _customSymbols, CompilationProject);
+            fileCompiler = new FileCompiler(initialTextDocumentLines, compilationProjet.SourceFileProvider, compilationProjet, compilationProjet.CompilationOptions, _customSymbols, compilationProjet);
 #endif
 
-            lock (_lockForOpenedDocuments)
+            if (workspaceProject.RemoveDocument(docContext))
             {
-                if (_openedDocuments.ContainsKey(docContext.Uri))
-                    CloseSourceFile(docContext.Uri); //Close and remove the previous opened file.
-
-                _openedDocuments.Add(docContext.Uri, docContext);
+                CloseSourceFile(docContext.Uri); //Close and remove the previous opened file.
             }
+            workspaceProject.AddDocument(docContext);
```

Comment:
Changed in [e1c296c](https://github.com/TypeCobolTeam/TypeCobol/pull/2145/commits/e1c296c5835196790db878e8efdab69f6a196c18)

But I renamed it TryCloseSourceFile, because it is an attempt to close the document if it is aready opened.